bottlecap/src/bin/bottlecap/main.rs

```diff
@@ -504,6 +504,10 @@ async fn extension_loop_active(
                     if !flush_control.should_flush_end() {
                         break;
                     }
+
+                    if matches!(flush_control.flush_strategy, FlushStrategy::Periodically(_)) {
+                        break;
+                    }
```

Comment:
Not sure I understand this

Reply:
woops, needed to remove the above. We were repurposing `should_flush_end()` to effectively mean `is the flush strategy periodically), so I'm just hoisting that up here.

This code was effectively:
```
if !!matches!(flush_control.flush_strategy, Periodically)
```